yarn-project/circuit-types/src/logs/l1_payload/l1_note_payload.ts

```diff
@@ -98,6 +118,12 @@ export class L1NotePayload extends L1Payload {
       EncryptedNoteLogIncomingBody.fromCiphertext,
     );
 
+    if (!isNoteTypeIdValid(incomingBody.noteTypeId)) {
```

Comment:
Do we need to have the `isNoteTypeIdValid` or could we just use the constructor of a `NoteTypeId` directly if it is something specific similar to a selector etc. Instead of just using the `Fr` that we are right now. It is slightly odd to have a `Fr` but then have extra conditions that we must remember to apply instead of just a small class 🤷